compiler/modules/bir/verify.bal

```diff
@@ -184,7 +184,11 @@ function verifyListConstruct(VerifyContext vc, ListConstructInsn insn) returns E
     }
     Operand[] operands = insn.operands;
     foreach int i in 0 ..< operands.length() {
-        check verifyOperandType(vc, operands[i], t:listAtomicTypeMemberAt(lat, i), "type of list constructor member is not allowed by the list type", insn.pos);
+        SemType st = t:listAtomicTypeMemberAt(lat, i);
```

Comment:
Let's deal with the unhelpful error message in a separate PR.

It's better to leave this as is and deal with this expr.bal: we can give a proper location as we did with function arguments.